bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/util/AbstractFormComponentImpl.java

```diff
@@ -477,20 +482,43 @@ public boolean isEmpty(SerializerProvider provider, String value) {
         }
     }
 
+    private boolean isAllowedType(Object value) {
+        return value instanceof String || value instanceof String[] || value instanceof Boolean || value instanceof Boolean[]
```

Comment:
This can be simplified using streams,

```
  return Stream.of(
            String.class, String[].class,
            Boolean.class, Boolean[].class,
            Calendar.class, Calendar[].class,
            BigDecimal.class, BigDecimal[].class,
            Long.class, Long[].class
    ).anyMatch(type -> type.isInstance(value));
    ```